### PR TITLE
[To rel/0.12] [ISSUE-3805]  OOM caused by Chunk cache

### PR DESCRIPTION
**Pull request by Alima777**
We used  to fuzzy computing to calculate the cache size, which led to some large caches being incorrectly counted, resulting in oom. And some concurrent problems may be caused by this.

Therefore, we try to calculate the cache size exactly to avoid these problems.

### CODE REVIEW

**Comment by Alima777** [comment]
We need to test the performance comparing with the previous.

**Comment by HTHou** [comment]
Plz also update the release note.

File: server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkCache.java [hunk 1]
```diff
@@ -63,50 +62,28 @@ private ChunkCache() {
...
-                      count = 1;
-                      currentSize = averageSize;
-                      entryAverageSize.set(averageSize);
-                    }
-                    return currentSize;
-                  }
-                })
+                      if (entrySize > 1024 * 1024 * 10) {
+                        logger.warn(String.format("Chunk size is %d over 10M", entrySize));
+                      }
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkCache.java:73]:
```suggestion
```

Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkCache.java:73]:
```suggestion
```

**Review by qiaojialin — APPROVED** [review]

**Comment by HTHou** [comment]
Don't forget to cherry pick it to master.

**Pull request merged**